system-helper/flatpak-system-helper.c

```diff
@@ -403,7 +405,7 @@ handle_deploy (FlatpakSystemHelper   *object,
 
   g_debug ("Deploy %s %u %s %s %s", arg_repo_path, arg_flags, arg_ref, arg_origin, arg_installation);
 
-  system = dir_get_system (arg_installation, get_sender_pid (invocation), &error);
+  system = dir_get_system (arg_installation, get_sender_pid (invocation), (arg_flags & FLATPAK_HELPER_DEPLOY_FLAGS_NO_INTERACTION), &error);
```

Comment:
I suspect some of these are passing a value other than 0 or 1 to a `gboolean` argument? Or, at least, they're using a pattern that would make it easy for that to happen, which doesn't seem good.

I think I prefer the pattern seen in `flatpak_authorize_method_handler()` where we assign to a temporary `gboolean`. Or, failing that, throw in a `!= 0`.

Reply:
> I suspect some of these are passing a value other than 0 or 1 to a `gboolean` argument

They are, although that shouldn’t matter unless some later code is doing an explicit `== TRUE` comparison, no?

Reply:
The documentation says

> Variables of this type should only contain the value TRUE or FALSE

I think Postel's principle implies we should always set `gboolean` variables to `TRUE` or `FALSE`, but test them in ways that would cope with a larger "truthy" value (`if (foo)` rather than `if (foo == TRUE)`).

Reply:
Updated to do that